### PR TITLE
limit concurrent cull requests

### PR DESCRIPTION
**Pull request by minrk**
avoids simultaneous request for deletion of lots of users, which can slow down the Hub

### CODE REVIEW

File: examples/cull-idle/cull_idle_servers.py [hunk 1]
```diff
@@ -90,7 +91,21 @@ def cull_idle(url, api_token, inactive_limit, cull_users=False, max_age=0):
...
     now = datetime.now(timezone.utc)
     client = AsyncHTTPClient()
-    resp = yield client.fetch(req)
+
+    if concurrency:
+        semaphore = Semaphore(concurrency)
+        @coroutine
+        def fetch(req):
+            """client.fetch wrapped in a semaphore to limit concurrency"""
+            yield semaphore.acquire()
```
Comment (Carreau) [inline examples/cull-idle/cull_idle_servers.py]: Can we use `async with`  or is it 3.6+ only ?

Comment (Carreau) [reply examples/cull-idle/cull_idle_servers.py]: Nah, 3.6+ never mind. Have been running 3.6 for too long.

**Comment by betatim** [comment]
LGTM.

One kind of related question: there are a few copies of the culler script floating around (here and th z2jh guide). Is it worth thinking about distributing a service like this as part of juypterhub so we can reduce the number of copies? Was thinking about this because this would be useful for mybinder.org as well.

**Comment by minrk** [comment]
@betatim yeah, there are tradeoffs.

1. make it part of JupyterHub
  - pro: distributed with jupyterhub, always compatible with jupyterhub
  - con: cannot distribute fixes without making jupyterhub releases
2. make it a standalone package
  - pro: easy to reuse in multiple places
  - con: another package to maintain/distribute

At this point, making it a simple lightweight package is probably the thing to do.

**Comment by minrk** [comment]
Follow-up, in case it wasn't clear already: we *do* use this in mybinder.org, since it is inherited from z2jh.

**Comment by betatim** [comment]
My confusion comes about because there is https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/master/images/hub/cull_idle_servers.py and the copy in this repo. So right now I think: we use `cull_idle_servers.py` in mybinder.org but we use the version from the z2jh repo (not the version from this repository).

**Comment by minrk** [comment]
@betatim exactly right, yes. I think a standalone package is where we should end up. For now, I'll copy this script a few more times before one of us gets bothered enough to make it a real package.

**Pull request merged**